lambda-source/index.js

```diff
@@ -62,18 +63,49 @@ exports.handler = async (event, context, callback) => {
           success_location: REDIRECT_URL,
           error_location: REDIRECT_URL
         }
-      })
-    }
+      }
+    )
+    return data;
   } catch (err) {
-    console.error("Error: ", err);
+    console.error("Form Error: ", err);
+    return false;
   }
+}
 
-  let redirectURL = REDIRECT_URL;
-  var response = {
-    "statusCode": 302,
-    "headers": {
-      Location: redirectURL,
-    }
-  };
-  callback(null, response);
-};
\ No newline at end of file
+const Responses = {
+  Unauthorized: () => ({ statusCode: 401 }),
+  BadRequest: () => ({ statusCode: 400 }),
+  Redirect: (Location) => ({ statusCode: 302, headers: { Location }})
+}
+
+async function proxyForm(event, _context) {
+  const buff = Buffer.from(event.body, 'base64');
+  const text = buff.toString('ascii');
+  const recaptchaPair = text.split('&').find(pair => pair.startsWith('g-recaptcha-response'));
+  
+  if (!recaptchaPair) {
+    // reCaptcha not provided at all
+    // maybe bad request?
+    return Errors.Unauthorized()
+  }
+  
+  const [_key, captchaResponse] = recaptchaPair.split("=");
+  console.log("Provided reCaptcha value:", captchaResponse)
+  
+  const isRecaptchaValid = await verifyRecaptcha(captchaResponse);
+  if (!isRecaptchaValid) {
+    return Errors.Unauthorized()
+  }
+
+  const formResponse = await sendForm(text);
+  if (!formResponse) {
+    return Errors.BadRequest()
+  }
+
+  return Responses.Redirect(REDIRECT_URL);
+}
+
+exports.handler = async function handler(event, context) {
+  console.log("Full event body", event.body);
+  return proxyForm(event, context);
+};
```

Comment:
for some reason it's not liking something about the wiring up of this handler:

![image](https://user-images.githubusercontent.com/7630696/155247294-c795a4ed-2055-4345-8858-c9d4e3e04d9a.png)

Reply:
Story of my life.  I'll check it out.

Reply:
Didn't iron out this specifically yet; it could be an issue with the `zip` script I put together q&d or simply how I changed the handler.  I'll mess with it more later, but I fixed the other stuff at least.